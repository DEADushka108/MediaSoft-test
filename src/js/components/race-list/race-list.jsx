import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {ActionCreator} from '../../store/ships/ships';
import {connect} from 'react-redux';
import {getUniqueRaceList} from '../../utils/utils';
import {getActiveRace, getShipsList} from '../../store/ships/selectors';

const RaceList = (props) => {
  const {shipsList, currentRace, onRaceClick} = props;
  const uniqueRaceList = getUniqueRaceList(shipsList);
  const [activeRace, setActiveRace] = useState(currentRace);

  return <ul className="catalog__race-list">
    {uniqueRaceList.map((race, index) => {
      return <li key={`${race}-${index}`} className={`catalog__race-item ${(activeRace === race) ? `catalog__race-item--active` : ``}`} onClick={() => {
       onRaceClick(race);
       setActiveRace(race); 
      }}>
        <a className="catalog__race-link">{race}</a>
      </li>;
    })}
  </ul>;
};

RaceList.propTypes = {};

const mapStateToProps = (state) => ({
  shipsList: getShipsList(state),
  currentRace: getActiveRace(state),
});

const mapDispatchToProps = (dispatch) => ({
  onRaceClick(race) {
    dispatch(ActionCreator.setActiveRace(race));
  },
})

export {RaceList};
export default connect(mapStateToProps, mapDispatchToProps)(RaceList);