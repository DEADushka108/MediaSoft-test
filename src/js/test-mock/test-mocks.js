export * from './const/const';
export * from './ships/ships';
export * from './store/store';
