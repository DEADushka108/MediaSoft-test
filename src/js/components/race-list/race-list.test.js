import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {shipsList, store, FilterSettings, noop} from '../../test-mock/test-mocks';
import {RaceList} from './race-list';

it(`Should RaceList component render correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <RaceList
          shipsList={shipsList}
          currentRace={FilterSettings.DEFAULT_VALUE}
          onRaceClick={noop}
        />
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
