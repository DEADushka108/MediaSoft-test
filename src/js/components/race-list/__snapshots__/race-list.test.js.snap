// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should RaceList component render correctly 1`] = `
<ul
  className="catalog__race-list"
>
  <li
    className="catalog__race-item catalog__race-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__race-link"
    >
      All
    </a>
  </li>
  <li
    className="catalog__race-item "
    onClick={[Function]}
  >
    <a
      className="catalog__race-link"
    >
      Alien
    </a>
  </li>
  <li
    className="catalog__race-item "
    onClick={[Function]}
  >
    <a
      className="catalog__race-link"
    >
      Human
    </a>
  </li>
  <li
    className="catalog__race-item "
    onClick={[Function]}
  >
    <a
      className="catalog__race-link"
    >
      Mixed
    </a>
  </li>
</ul>
`;
